### PR TITLE
Business logic added

### PR DESCRIPTION
**Pull request by mikolajmikolajczyk**
## Summary by CodeRabbit

- **New Features**
	- Enhanced GitHub action 'semver-sugar' with new input parameters for improved release management.
	- Introduced outputs for tracking created tags and increment types.
	- Added a versioning system based on semantic versioning principles with functionality for version increments.
	- Added utility functions for interacting with GitHub's API, including client creation and tag retrieval.
	- Comprehensive unit tests for release execution logic and version management functions.
	- New GitHub Actions workflow for unit testing across multiple operating systems.

- **Bug Fixes**
	- Removed unnecessary sleep delay functionality for a more streamlined operation.

- **Documentation**
	- Updated branding information including an icon and color for better visibility.
	- Updated Go version in the workflow configuration for improved build performance.
	- Removed the test workflow configuration for clearer execution.

### CODE REVIEW

File: pkg/utils/github.go [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+		return "", err
+	}
+	expectedRange, err := semver.ParseRange(versionRange)
+	if err != nil {
+		return "", err
+	}
+
+	latest := semver.MustParse("0.0.0")
+	for _, ref := range refs {
+		version, err := semver.ParseTolerant(strings.Replace(*ref.Ref, "refs/tags/", "", 1))
```
[the remark this answers is not in the document]
Comment (mikolajmikolajczyk) [reply pkg/utils/github.go:75]: Can't do that as semver name would interfere with existing package I have.

File: .github/workflows/build-dist.yml [hunk 2]
```diff
@@ -55,13 +59,21 @@
...
           git config user.email "github-actions[bot]@users.noreply.github.com"
           git config user.name "github-actions[bot]"
           git add "dist/main_*"
           if git diff --cached --quiet; then
             echo "No changes to commit"
           else
             git commit -m "Build new distributable packages for version ${{ github.sha }}"
-            git push origin "${{ github.ref }}"
+            echo "::set-output name=changes_exist::true"
           fi
```
[the remark this answers is not in the document]
Comment (wbed-c8y) [reply .github/workflows/build-dist.yml:73]: +1

File: pkg/utils/github.go [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+	event, ok := parsed.(*github.PullRequestEvent)
+	if !ok {
+		return nil, fmt.Errorf("invalid event")
+	}
+	return event, nil
+
+}
+
+func (impl *GithubActionImpl) GetGithubLatestTag(versionRange string) (string, error) {
+	core.Infof("Getting latest tag from github repositoryasdasd %v", impl.Repository)
```
[the remark this answers is not in the document]
Comment (wbed-c8y) [reply pkg/utils/github.go:59]: +1 :D

File: main.go [hunk 4]
```diff
@@ -1,28 +1,194 @@
...
+		core.Info("Getting next tag from latest tag and increment type")
+		nextTag, err := ghActionIface.GetNextTag(latestTag, actionConfig.Increment, actionConfig.TagFormat)
+		if err != nil {
+			core.Error(err.Error())
+			Exit(1)
+		}
+		core.Info("Next tag is: " + nextTag)
+		actionConfig.NextTag = nextTag
+	}
+
```
[the remark this answers is not in the document]
Comment (mikolajmikolajczyk) [reply main.go]: This would make code harder to read.

File: pkg/utils/github.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (mikolajmikolajczyk) [reply pkg/utils/github.go]: Not applicable anymore

**Pull request merged**